src/Microsoft.CodeQuality.Analyzers.Exp/Core/FlowAnalysis/Framework/DataFlow/MapAbstractDomain.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Operations.DataFlow
+{
+    /// <summary>
+    /// An abstract domain implementation for analyses that store dictionary typed data.
+    /// </summary>
+    internal class MapAbstractDomain<TKey, TValue> : AbstractDomain<IDictionary<TKey, TValue>>
+    {
+        private AbstractDomain<TValue> _valueDomain;
+
+        public MapAbstractDomain(AbstractDomain<TValue> valueDomain)
+        {
+            _valueDomain = valueDomain;
+        }
+
+        public override IDictionary<TKey, TValue> Bottom => new Dictionary<TKey, TValue>();
+
+        public override int Compare(IDictionary<TKey, TValue> oldValue, IDictionary<TKey, TValue> newValue)
```

Comment:
>Compare [](start = 28, length = 7)

this one does not work like a regular Compare:
Compare(a,b) == -Compare(b,a)

Reply:
Thanks, I will take a look. This file is directly ported from Edgardo's prototype, I will understand the logic further.